### DISCUSSION TITLE
Processos e Gerenciamento de projetos

### DISCUSSION THREAD

**Discussion by Cleysonlb** [frontendbr/forum#351] [Discussão]
Bom dia galera blz?! Bom, queria levantar esse ponto aqui e discutir sobre, vocês usam algum processo para gerenciar seus projetos, Existe algumas metodologias que posso buscar para implementar na empresa em que trabalho?!

exemplo de tópicos:
* padrão de versionamento
* padrão de código
* documentação
* como o projeto pode estar para não existir uma barreira tão grande para quem entrar agora como desenvolvedor?

**Comment by herberthenrique** [comment]
Cara,
Eu sou um grande apaixonado pelo git flow, aqui tem algumas explicações muito boas sobre o flow e seu uso.
https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow
https://fjorgemota.com/git-flow-uma-forma-legal-de-organizar-repositorios-git/

Isso sem contar que existem algumas ferramentas pra te ajudar nesse processo.
Como esse: http://danielkummer.github.io/git-flow-cheatsheet/index.pt_BR.html

Lembrando de algo super importante: não existe bala de prata e o flow sempre pode ser adaptado.

---

Quanto a padrões de código, eu uso o [airbnb](https://github.com/airbnb/javascript) para javascript. Se você quer implementar um padrão na sua empresa é sempre bom começar e depois ir definindo as regras de acordo com o que vocês acreditam depois.
E para ajudar a validar o padrão eu uso o [ESLINT](http://eslint.org/) e configuro pré commit hooks no git com o [husky](https://github.com/typicode/husky).

---

Documentação agora é algo mais chatinho, eu diria.  Depende muito do software que vocês criam, mas se for pra começar, eu recomendo um README bem escrito, você pode tentar algo como o [Aewsome README](https://github.com/matiassingers/awesome-readme).
Eu já usei bastante o [swagger](http://swagger.io/) também.

Só lembrando que apenas comentar não é documentação.   [Esse texto aqui explica a razão](http://apdevblog.com/comments-in-code/)

**Comment by vitocaio** [comment]
Bom existe vários métodos de gerenciamento de Projetos, uma forma bem comum é a utilização de metodologias ágeis, da uma olhada em uma bastante comum, chamada scrum. simples e fácil de  de implementar

**Comment by herberthenrique** [comment]
Ah, sobre novas pessoas novas no projeto.
Acho que isso é mais uma questão de cultura, quando entram novas pessoas eu sempre recomendo trocar uma ideia com a pessoa antes, sobre o sistema e a arquitetura e bastante pair programming nos primeiros dias. :)
